cvat/apps/dataset_manager/bindings.py

```diff
@@ -188,20 +191,76 @@ def _export_attributes(self, attributes):
         return exported_attributes
 
 class CommonData(InstanceLabelData):
-    Shape = namedtuple("Shape", 'id, label_id')  # 3d
-    LabeledShape = namedtuple(
-        'LabeledShape', 'type, frame, label, points, occluded, attributes, source, rotation, group, z_order, elements, outside, id')
-    LabeledShape.__new__.__defaults__ = (0, 0, 0, [], False, None)
-    TrackedShape = namedtuple(
-        'TrackedShape', 'type, frame, points, occluded, outside, keyframe, attributes, rotation, source, group, z_order, label, track_id, elements, id')
-    TrackedShape.__new__.__defaults__ = (0, 'manual', 0, 0, None, 0, [], None)
-    Track = namedtuple('Track', 'label, group, source, shapes, elements, id')
-    Track.__new__.__defaults__ = ([], None)
-    Tag = namedtuple('Tag', 'frame, label, attributes, source, group, id')
-    Tag.__new__.__defaults__ = (0, None)
-    Frame = namedtuple(
-        'Frame', 'idx, id, frame, name, width, height, labeled_shapes, tags, shapes, labels, subset')
-    Label = namedtuple('Label', 'id, name, color, type')
+    class Shape(NamedTuple):
+        id: int
+        label_id: int
+
+    class LabeledShape(NamedTuple):
+        type: int = 0
+        frame: int = 0
+        label: int = 0
+        points: Sequence[int] = ()
+        occluded: bool = False
+        attributes: dict | None = None
+        source: str | None = None
+        rotation: float = 0.0
+        group: int = 0
+        z_order: int = 0
+        elements: Sequence[CommonData.LabeledShape] = ()
+        outside: bool = False
+        id: int | None = None
+
+    class TrackedShape(NamedTuple):
+        type: int = 0
+        frame: int = 0
+        points: Sequence[int] = ()
+        occluded: bool = False
+        outside: bool = False
+        keyframe: bool = False
+        attributes: Sequence[int] = ()
```

Comment:
I'm not sure the added defaults are useful, especially tuples. How do you think they should be used?

Reply:
Regarding added defaults, you're right. I've removed ones that weren't there before.

As for tuples, they serve as a replacement for lists that were in place before. 

You never want to assign a mutable collection as a default attribute:

```py
class Test(NamedTuple):
    attr: list[str] = []


first_instance = Test()
second_instance = Test()

print(first_instance is not second_instance)  # True
print(first_instance.attr is second_instance.attr)  # True
first_instance.attr.append("foo")
print(first_instance, second_instance)  # Test(attr=['foo']) Test(attr=['foo'])
```